bcrypt and updated pom.xml

edimusrex opened the pull request:
Added component for bcrypt, also update the pom.xml with the appropriate Maven repositories

edimusxero committed:
bcrypt

Added component for bcrypt, also update the pom.xml with the appropriate Maven repositories

tmortagne commented on src/main/java/org/mindrot/jbcrypt/BCrypt.java:
```diff
@@ -0,0 +1,777 @@
+// Copyright (c) 2006 Damien Miller <djm@mindrot.org>
```
I don't like this too much. I'm sure this can be found in some library in Maven central, Bouncy Castle seems to have support for BCrypt for example ?

tmortagne commented on pom.xml:
```diff
@@ -1,90 +1,100 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 
-<!--
```
This file is supposed to have a header.

tmortagne commented on src/main/java/org/xwiki/contrib/authentication/jdbc/internal/BcryptPasswordHasher.java:
```diff
@@ -0,0 +1,24 @@
+package org.xwiki.contrib.authentication.jdbc.internal;
```
This file should have the same header than other Java files (should be just a copy/paste).

tmortagne commented:
Thanks a lot for that contribution but I really don't like embedding code from somewhere else. You should use some library like bouncy castle (or whatever library where this BCrypt implementation come from) instead and just add it to the dependencies in the pom.xml.

edimusrex commented:
ok, I'll look into these changes.  I am new to git hub and java but thought I would share.

edimusxero committed:
added headers

added headers

edimusxero committed:
updated pom

removed bcrypt class and added it to pom dependency, includes library in compiled jar upon build.

tmortagne commented on pom.xml:
```diff
@@ -1,90 +1,150 @@
 
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-  <parent>
```
You should not remove the parent.

tmortagne commented on pom.xml:
```diff
@@ -1,90 +1,150 @@
+                
+    </dependencies>
+    
+    <build>
```
As I said you should not embbed dependencies, the Extension Manager will deal with dependencies.

The pom.xml was fine before, you should probably just revert all the changes you made to this file and simply add your <dependency> since everything else does not have much to do with adding support for bcrypt.

tmortagne commented:
Applied your pull request and will fix the pom.xml.

tmortagne commented:
1.1 release done, see https://forum.xwiki.org/t/jdbc-authenticator-1-1-released

Thanks !
